src/components/ContentAccordion.vue

```diff
@@ -0,0 +1,197 @@
+<template>
+  <div>
+    <button
+      class="expansion-button"
+      :aria-controls="'expansion-content-' + ariaTitle"
+      :id="'expansion-control-' + ariaTitle"
+      @click="togglePanel"
+    >
+      <div class="w-full">
+        <div class="accordion-title pb-1">
+          <slot name="title">
+            {{ title }}
+          </slot>
+        </div>
+        <div class="accordion-description pb-1" v-if="description">
+          <slot name="description">
+            {{ description }}
+          </slot>
+        </div>
+        <div class="flex items-center">
+          <span class="accordion-description">
+            {{ showPanel ? hideText : showText }}
+          </span>
+          <span class="icon-wrapper">
+            <span
+              class="panel-icon"
+              :class="{
+                'show-panel-icon': showPanel,
+                'hide-panel-icon': !showPanel
+              }"
+            ></span>
+          </span>
+        </div>
+      </div>
+    </button>
+    <transition name="slide-fade">
+      <div
+        class="panel-content"
+        :aria-hidden="!showPanel"
+        :id="'content-' + ariaTitle"
+        v-if="showPanel"
+      >
+        <div class="inner-panel">
+          <slot></slot>
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+
+export default defineComponent({
+  name: "wra-content-accordion",
+  props: {
+    title: {
+      type: String as PropType<string>,
+      required: true
+    },
+    /**
+     * Description text displayed below the title
+     */
+    description: {
+      type: String as PropType<string>,
+      required: false
+    },
+    /**
+     * A unique string used to generate ARIA attributes for accessibility.
+     */
+    ariaTitle: {
+      type: String as PropType<string>,
+      required: true
+    },
+    /**
+     * Text to display when the panel is collapsed
+     */
+    showText: {
+      type: String as PropType<string>,
+      default: "Show"
+    },
+    /**
+     * Text to display when the panel is expanded
+     */
+    hideText: {
+      type: String as PropType<string>,
+      default: "Hide"
+    }
+  },
+  data() {
+    return {
+      showPanel: false as boolean
```

Comment:
```suggestion
      showPanel: false
```
Cast is not needed as we are setting to `false`